transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -302,11 +309,13 @@ public void run() {
     @Override
     protected void run() {
         for (;;) {
-            boolean oldWakenUp = wakenUp.getAndSet(false);
             try {
-                if (hasTasks()) {
-                    selectNow();
-                } else {
+                if (selectNow() == 0 && !hasTasks()) {
```

Comment:
What I was originally thinking about for https://github.com/netty/netty/pull/4955#discussion_r55561206 was something like the following:

```java
public interface BackoffStrategy {
    boolean delaySelect(int selectNowCount, boolean hasTasks);
}

// Then the call site here would look like
for (;;) {
  try {
    if (backoff.delaySelect(selectNow(), hasTasks())) {
      continue;
    }

    boolean oldWakenUp = wakenUp.getAndSet(false);
    select(oldWakenUp);
...
```

This way the BackoffStrategy has more visibility and can apply more complicated strategies (weighted average, consecutive empties increase the delay, delay until there are x amount of tasks ready, variable time delay based upon how many tasks are ready, etc...). Does that make sense?